src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -261,15 +262,25 @@ public void write() {
               }
             } else {
               SinkRecord projectedRecord = SchemaUtils.project(record, currentSchema, compatibility);
-              writeRecord(projectedRecord);
-              buffer.poll();
-              if (shouldRotate(now)) {
+              String partition = partitioner.encodePartition(projectedRecord);
+              if (currentPartition != null && !currentPartition.equals(partition)) {
+                currentPartition = partition;
```

Comment:
Yeah @13h3r that's what I was thinking. Although actually using a cron expression doesn't seem great since people often struggle to get the expression they want correct. An alternative would be to still use `rotate.schedule.ms` but add another config that does something like control alignment, e.g. it forces the intervals to start at 00:00:00 of each day.

One thing to be careful of here is also time synchronization between different systems. This actually may be an issue with the existing code as well... If you move a task, you don't want it to overwrite a file that some other node already wrote just because the clocks of the two systems are not perfectly synchronized.